newdle/api.py

```diff
@@ -216,3 +217,25 @@ def update_participant(args, code, participant_code):
         setattr(participant, key, value)
     db.session.commit()
     return ParticipantSchema().jsonify(participant)
+
+
+@api.route('/newdle/<code>/participants', methods=('POST',))
+@allow_anonymous
+@use_args(CreateAnonymousParticipantSchema(), locations=('json',))
+def create_anonymous_participant(args, code):
+    newdle = Newdle.query.filter_by(code=code).first_or_404('Invalid code')
+    # Once again, we can't validate this in webargs
```

Comment:
You could use this API just to create the participant (ie not accepting answers) and then submit their answers in a a separate request. That way you don't need tu duplicate this rather big piece of code.

But I guess with the hook that's a bit harder to implement...

Reply:
Good idea.

Reply:
Done